File: lua/ch06/integral.lua
-- Exercise 6.1: Write a function integral that receives a function
-- f and returns an approximation of its integral

function integral(func, a, b)
    local sum = 0
    local detX = 1e-6
    for i = a, b, detX do
        sum = sum + detX * func(i)
    end
    return sum 
end

function f(x)
    return x * x;
end

print(integral(f, 1, 2))

File: lua/ch06/integral.c
#include <stdio.h>

float f(float x)
{
    return x * x;
}
int main()
{
    double sum = 0.0;
    double det = 1e-9;
    printf("det:%e\n", det);
    for(double i  = 1.0; i <=2.0; i = i + det)
    {
        sum = sum + det * f(i); 
    }
    printf("sum:%f\n", sum);
}